### PR TITLE
Allow uppercase characters in header names

### PR DESCRIPTION
**Pull request by ijsnellf**
Fixes https://github.com/istio/istio/issues/7859

### CODE REVIEW

**Comment by rshriram** [comment]
also to release-1.0

File: istioctl/cmd/istioctl/testdata/v1alpha3/merged-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+apiVersion: networking.istio.io/v1alpha3
```
Comment (ZackButcher) [inline istioctl/cmd/istioctl/testdata/v1alpha3/merged-gateway.yaml]: Should these be part of the PR? I think they're generated by a test, I _thought_ I added them to gitignore a while back...

Comment (ijsnellf) [reply istioctl/cmd/istioctl/testdata/v1alpha3/merged-gateway.yaml]: my bad

**Pull request merged**